GDSCommon-Demo/GDSCommon-Demo/Mocks/MockTextFieldViewModel.swift

```diff
@@ -0,0 +1,20 @@
+@testable import GDSCommon
+import UIKit
+
+struct MockTextFieldViewModel<InputType>: TextFieldViewModel {
+    typealias InputType = Bool
```

Comment:
This `InputType` Generic doesn't seem to be mentioned in the rest of this Type and seems to be consistently a `Bool` in other files. Am I missing something here? Could it not just be a Bool and we don't need a Generic at all?

Reply:
It's only for the mock for testing that it is a `Bool`. It wouldn't typically make much sense to be a `Bool` for normal use though I don't think?

For example, on the `DaysPerWeekTextFieldViewModel` it is set to a `Double`; that's the type that we are expecting / validating against.

I can see in the future this screen needing to be configurable so that developers can set when the validation occurs. For example, right now it happens on every change of the text field input. But this would prevent entering more complex patterns where multiple characters are needed prior to validation. In those scenarios, it might be that validation should only occur when tapping done on the keyboard for example.

That future flexibility would be highly beneficial for validating different types. But is beyond the scope of what was required here.